### PR TITLE
RSA Object update for token function

### PR DESCRIPTION
**Pull request by kylecweeks**
Allow for accepting RSA object into the function instead of building it within.

Additional minor where to where-object update.

Documentation updated.

### CODE REVIEW

File: UMN-Google.psm1 [hunk 1]
```diff
@@ -82,27 +82,33 @@ function ConvertTo-Base64URL
...
                     Get-GOAuthTokenService -scope "https://www.googleapis.com/auth/spreadsheets" -certPath "C:\users\$env:username\Desktop\googleSheets.p12" -certPswd 'notasecret' -iss "serviceAccount@googleProjectName.iam.gserviceaccount.com"
 
+                .EXAMPLE
+                    Get-GOAuthTokenService -rsa $rsaSecurityObject -scope "https://www.googleapis.com/auth/spreadsheets" -iss "serviceAccount@googleProjectName.iam.gserviceaccount.com"
+
             #>
             [CmdletBinding()]
             Param
             (
-                [Parameter(Mandatory)]
```
Comment (FISHMANPET) [inline UMN-Google.psm1]: I'd use ParameterSets here, so one parameter set for the file with certpath, certpswd, and iss mandatory, and another parameterset for the cert as an object

File: UMN-Google.psm1 [hunk 2]
```diff
@@ -136,11 +142,13 @@ function ConvertTo-Base64URL
 
                 ################# Create JWT
                 # Prep JWT certificate signing
-                $googleCert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2($certPath, $certPswd,[System.Security.Cryptography.X509Certificates.X509KeyStorageFlags]::Exportable ) 
-                $rsaPrivate = $googleCert.PrivateKey 
-                $rsa = New-Object System.Security.Cryptography.RSACryptoServiceProvider 
-                $null = $rsa.ImportParameters($rsaPrivate.ExportParameters($true))
-                
+                if(!$rsa){
```
Comment (FISHMANPET) [inline UMN-Google.psm1]: with parameter sets you get `$PSCmdlet.ParameterSetName` so you could replace this with `$PSCmdlet.ParameterSetName -eq "CertFile"`

File: UMN-Google.psm1 [hunk 3]
```diff
@@ -82,30 +82,40 @@ function ConvertTo-Base64URL
...
+                [Parameter(Mandatory,ParameterSetName='Certificate')]
                 [string]$certPswd,
 
-                [Parameter(Mandatory)]
-                [string]$iss,
-                
-                [Parameter(Mandatory)]
-                [string]$scope
+                [Parameter(Mandatory,ParameterSetName='RSA')]
+                $rsa
```
Comment (FISHMANPET) [inline UMN-Google.psm1]: Not sure what will happen if you pass some nonsense in as $rsa, should probably do something like `[RSACryptoServiceProvider]$rsa` to ensure an RSA object is actually passed in.

**Comment by FISHMANPET** [comment]
ok, made the changes we discussed about passing in a cert object.  Also updated comment based help for the function and updated all the help docs.

I used your certTest runbook and validated that this behaves properly when passing in the cert object from Get-AutomationCertificate

**Comment by kylecweeks** [comment]
Reviewed Peter's in line edits. Looks good for cert objects as well.

**Pull request merged**